javascript/wordcount.js

```diff
@@ -1,51 +1,44 @@
 "use strict";
-var readline = require('readline');
 var TimSort = require('timsort');
 
 process.stdin.setEncoding('utf8');
-
-let rl = readline.createInterface({ input: process.stdin, terminal: false });
-let wordCounts = {};
-const regExp = /[ \t\n\r]+/g;
-
-rl.on('line', (line) => {
-  let words = line.trim().split(regExp);
-
-  for(let i = 0, len = words.length; i < len; i++) {
-    let word = words[i];
-
-    if (!word)
-      return;
-
-    if(wordCounts.hasOwnProperty(word)) {
-      wordCounts[word]++;
-    }
-    else {
-      wordCounts[word] = 1;
+const regExp = /[ \t\n\r]+/;
+var lastChunk = '';
+var counts = new Map();
+process.stdin.on('data', function (chunk) {
+  var words = chunk.split(regExp);
+  var count = words.length;
+  if (lastChunk) {
+    words[0] = lastChunk + words[0];
+  }
+  lastChunk = words[count - 1];
+  count--;
+  for (var i = 0; i < count; i++) {
+    var word = words[i];
+    if (word) {
+      counts.set(word, (counts.get(word) || 0) + 1);
     }
   }
-}).on('close', () => {
-  let wordList = Object.keys(wordCounts);
-
-  let compare = (x, y) => {
-    if(wordCounts[x] < wordCounts[y])
-      return 1;
-
-    if(wordCounts[x] === wordCounts[y] && x > y)
-      return 1;
-
-    return -1;
+}).on('end', function () {
+  if (lastChunk) {
+    counts.set(lastChunk, (counts.get(lastChunk) || 0) + 1);
   }
-
-  TimSort.sort(wordList, compare);
+  var res = [...counts.keys()];
```

Comment:
Which version of nodejs is this running on? Older versions don't have support for ES6 syntax and will error on this.